apps/app/src/pages/Repository/index.tsx

```diff
@@ -19,21 +19,32 @@ const RepositoryQuery = graphql(`
     repository(ownerLogin: $ownerLogin, repositoryName: $repositoryName) {
       id
       permissions
+      tests(first: 1, after: 0) {
+        pageInfo {
+          totalCount
+        }
+      }
```

Comment:
It is not perf enough. You should add `hasTests` query. Counting all tests is not required to see if there is one test.